src/facet_display/FacetDisplay.tsx

```diff
@@ -1,27 +1,27 @@
-import React, { useCallback } from "react"
+import React, { useMemo } from "react"
 import FilterableFacet from "./FilterableFacet"
 import Facet from "./Facet"
-import SearchFilter from "./SearchFilter"
+import FilterIndicator from "./FilterIndicator"
 import type {
   FacetManifest,
   Facets,
   Aggregation,
   Bucket,
-  BooleanFacets
+  BooleanFacets,
+  BucketWithLabel
 } from "./types"
 import { LEVELS, DEPARTMENTS } from "../constants"
-
-export type BucketWithLabel = Bucket & { label: string }
+import MultiFacetGroup from "./MultiFacetGroup"
 
 interface FacetDisplayProps {
-  facetMap: FacetManifest
+  facetManifest: FacetManifest
   /**
    * Returns the aggregation options for a given group.
    *
    * If `activeFacets` includes a facet with no results, that facet will
    * automatically be included in the facet options.
    */
-  facetOptions: (group: string) => Aggregation | null
+  facetOptions: Record<string, Bucket[]>
```

Comment:
This is one of the more significant changes. I made it for two reasons:

First, we want to group boolean facets together, **but they have different facet names**. For example, in MIT Open the boolean facets are configured as:
```ts
    // a MultiFacetGroupOptions
    {
      type: "group",
      facets: [
        { value: true, name: "free", label: "Free" },
        { value: true, name: "certification", label: "With Certificate" },
        { value: true, name: "professional", label: "Professional" },
      ],
    },
```
The component rendering this group of facets needs buckets for multiple facet names. The old `name => BucketWithLabels` prop format didn't work for this. The new format does.

Second: The new format seems simpler: We can just pass the aggregation object we get from OpenSearch. (We can still manipulate that object first if we need to remove some facet options for some reason).